server/methods/core/groups.app-test.js

```diff
@@ -0,0 +1,135 @@
+import { Meteor } from "meteor/meteor";
+import { check, Match } from "meteor/check";
+import { Factory } from "meteor/dburles:factory";
+import { expect } from "meteor/practicalmeteor:chai";
+import { sinon } from "meteor/practicalmeteor:sinon";
+import { Reaction } from "/server/api";
+import { Accounts, Groups } from "/lib/collections";
+import Fixtures from "/server/imports/fixtures";
+import { getUser } from "/server/imports/fixtures/users";
+
+Fixtures();
+
+describe("Group test", function () {
+  let methods;
+  let sandbox;
+  let shop;
+  let user;
+  const sampleGroup = {
+    name: "Shop Manager",
+    permissions: ["sample-role1", "sample-role2"]
+  };
+
+  before(function () {
+    methods = {
+      createGroup: Meteor.server.method_handlers["group/createGroup"],
+      addUser: Meteor.server.method_handlers["group/addUser"],
+      updateGroup: Meteor.server.method_handlers["group/updateGroup"],
+      removeUser: Meteor.server.method_handlers["group/removeUser"]
+    };
+  });
+
+  beforeEach(function () {
+    sandbox = sinon.sandbox.create();
+    shop = Factory.create("shop");
+    user = getUser();
+    // make the same user on Meteor.users available on Accounts
+    Accounts.upsert({ _id: user._id }, { $set: { userId: user._id } });
+  });
+
+  afterEach(function () {
+    Groups.remove({});
+    sandbox.restore();
+    Meteor.users.remove({});
+    Factory.create("shop");
+  });
+
+  function spyOnMethod(method, id) {
+    return sandbox.stub(Meteor.server.method_handlers, `group/${method}`, function () {
+      check(arguments, [Match.Any]); // to prevent audit_arguments from complaining
+      this.userId = id;
+      return methods[method].apply(this, arguments);
+    });
+  }
+
+  it("should create a group for a particular existing shop", function () {
+    sandbox.stub(Reaction, "hasPermission", () => true);
+    spyOnMethod("createGroup", shop._id);
+
+    Meteor.call("group/createGroup", sampleGroup, shop._id);
+    const group = Groups.findOne({ shopId: shop._id });
```

Comment:
Finding groups belonging to a shop can be done with the slug property. More can be seen in `addDefaultRoles.app-test.js`. This is just doing a findOne because just one group was added in the beforeEach.